app/events/admin.go

```diff
@@ -264,22 +282,38 @@ func (a *admin) directReport(update tbapi.Update, updateSamples bool) error {
 	}
 
 	// delete original message
-	if _, err := a.tbAPI.Request(tbapi.DeleteMessageConfig{ChatID: a.primChatID, MessageID: origMsg.MessageID}); err != nil {
+	if _, err := a.tbAPI.Request(tbapi.DeleteMessageConfig{BaseChatMessage: tbapi.BaseChatMessage{
+		MessageID:  origMsg.MessageID,
+		ChatConfig: tbapi.ChatConfig{ChatID: a.primChatID},
+	}}); err != nil {
 		errs = multierror.Append(errs, fmt.Errorf("failed to delete message %d: %w", origMsg.MessageID, err))
 	} else {
 		log.Printf("[INFO] spam message %d deleted", origMsg.MessageID)
 	}
 
 	// delete reply message
-	if _, err := a.tbAPI.Request(tbapi.DeleteMessageConfig{ChatID: a.primChatID, MessageID: update.Message.MessageID}); err != nil {
+	if _, err := a.tbAPI.Request(tbapi.DeleteMessageConfig{BaseChatMessage: tbapi.BaseChatMessage{
+		MessageID:  update.Message.MessageID,
+		ChatConfig: tbapi.ChatConfig{ChatID: a.primChatID},
+	}}); err != nil {
 		errs = multierror.Append(errs, fmt.Errorf("failed to delete message %d: %w", update.Message.MessageID, err))
 	} else {
 		log.Printf("[INFO] admin spam reprot message %d deleted", update.Message.MessageID)
 	}
 
+	var username string
```

Comment:
There is a similar code above with an ID as well. Maybe consider making a function for this.